src/scenes/ProgressReports/EditForm/Steps/SubmitReportStep/ConfirmVariantDialog.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Typography,
+} from '@mui/material';
+import { Role } from '~/api/schema/schema';
+import { PromptResponseListFragment } from '~/common/fragments';
+import { useSession } from '~/components/Session';
+import { ProgressReportEditFragment } from '../../ProgressReportEdit.graphql';
+
+interface ConfirmEmptyVariantsProps {
+  report: ProgressReportEditFragment;
+  open: boolean;
+  onContinue: () => void;
+  onCancel: () => void;
+  updateConfirmStatus: () => void;
+}
+
+export const ConfirmEmptyVariants = ({
+  report,
+  open,
+  onContinue,
+  onCancel,
+  updateConfirmStatus,
+}: ConfirmEmptyVariantsProps) => {
+  const { reportFile, teamNews, communityStories } = report;
+  const { session } = useSession();
+  const roles = new Set(session?.roles.value);
+  const hasReportProgressFile = !!reportFile.value;
+  const projectMgmtHeader = !hasReportProgressFile && reportFile.canEdit;
+  const hasSelectedTeamNewsPrompt = !!(teamNews.items.length > 0);
+  const hasSelectedStoryPrompt = !!(communityStories.items.length > 0);
+  const listItems = [];
+
+  const handleConfirmDialog = () => {
+    onContinue();
+  };
+
+  const handleCloseDialog = () => {
+    onCancel();
+  };
+
+  const getEmptyResponsesForRole = (
+    list: PromptResponseListFragment,
+    roles: Set<Role>
+  ) => {
+    if (list.items.length === 0) return [];
+    const responsesForRole = list.items[0]?.responses.filter((item) => {
+      return item.variant.responsibleRole
+        ? roles.has(item.variant.responsibleRole)
+        : false;
+    });
+    const emptyResponsesByRole = responsesForRole?.filter((item) =>
+      !item.response.value && item.response.canEdit ? true : false
```

Comment:
This can be shortened to `!item.response.value && item.response.canEdit` since the comparison itself returns a boolean. In general, there is never a need to make a ternary that ends with `true : false`. If coercion to a boolean is needed the !! operator can be used ie: `!!0 //false !!1 //true`